### DISCUSSION TITLE
Note regarding carrier IMEI whitelists

### DISCUSSION THREAD

**Discussion by flawedworld** [kyujin-cho/pixel-volte-patch#23] [Ideas]
Just so that some users are aware, some carriers may whitelist IMEIs of devices allowed to use VoLTE, which this cannot bypass. I know that Verizon in the USA is an example of this.

**Comment by MohabMohie** [comment]
any ideas how to change the device to one of the whitelisted IMEIs then?
iPhone 12 pro max, or Samsung Galaxy S22 for example?

can we use something out of this page?
https://deviceatlas.com/blog/list-of-user-agent-strings

**Reply by flawedworld** [reply]
Modern devices made by sane manufacturers will not let you change IMEI typically. You might be able to do it via some diagnostic APIs like on Qualcomm, but that is not persistent, and I don't know if that is even possible anymore.

User agents used in browsers are orthogonal to this issue of IMEI whitelisting. Rather the user agent used by IMS to communicate to the mobile network is typically device specific, and could also be responsible for whitelisting (But I doubt this is an issue for Pixels) but is itself, its own issue.

**Reply by MohabMohie** [reply]
I was able to get it working by calling my network operator. I just told them I bought a new phone and volte want working. They provisioned it from their side and it worked!

**Reply by aly-k** [reply]
@MohabMohie Are you in Egypt ? If so what's your carrier ?
